arangod/IResearch/AqlHelper.cpp

```diff
@@ -646,7 +646,7 @@ bool attributeAccessEqual(aql::AstNode const* lhs, aql::AstNode const* rhs,
 
   // TODO: is the "&" intionally. If yes: why?
   // cppcheck-suppress uninitvar; false positive
-  while (lhsValue.read(lhs, ctx) & rhsValue.read(rhs, ctx)) {
+  while (lhsValue.read(lhs, ctx) & (int)rhsValue.read(rhs, ctx)) {
```

Comment:
I don't like fix:
c style cast
signed integer operand with a binary bitwise operator

Reply:
Good points. Would
```suggestion
  while (lhsValue.read(lhs, ctx) &
         static_cast<unsigned>(rhsValue.read(rhs, ctx))) {
```
work for you? Alternatively, we could go more verbose like
```cpp
  bool lhsRead = lhsValue.read(lhs, ctx);
  bool rhsRead = rhsValue.read(rhs, ctx);
  while (lhsRead && rhsRead) {
    if (lhsValue != rhsValue) {
      return false;
    }

    lhs = lhsValue.root;
    rhs = rhsValue.root;
    lhsRead = lhsValue.read(lhs, ctx);
    rhsRead = rhsValue.read(rhs, ctx);
  }
  ```

Reply:
Alright thanks, I've applied the first suggestion in https://github.com/arangodb/arangodb/pull/16400/commits/7e0ed87cb12c58e3299d1a50adb9f0d11527eeb1.